db/db_impl/db_impl_open.cc

```diff
@@ -1545,6 +1545,27 @@ Status DB::Open(const DBOptions& db_options, const std::string& dbname,
                       !kSeqPerBatch, kBatchPerTxn);
 }
 
+Status DB::OpenAndTrimHistory(
+    const DBOptions& db_options, const std::string& dbname,
+    const std::vector<ColumnFamilyDescriptor>& column_families,
+    std::vector<ColumnFamilyHandle*>* handles, DB** dbptr,
+    const std::string& trim_ts) {
```

Comment:
We can change the type of `trim_ts` from `const std::string&` to `std::string` because we will eventually create a std::string inside RocksDB, i.e. in the background compaction job where compaction actually happens. By changing the type to `std::string` here, the applications does not have to worry about the lifecycle of this variable, making it easier for them. There will be no data copy overhead if we carefully use `std::move()` along the way until the variable gets moved to its final destination, e.g. `Compaction::trim_ts` and/or `CompactionJob::trim_ts`, etc.

Reply:
already change the type of trim_ts to std::string